docs/source-2.0/aws/aws-endpoints.rst

```diff
@@ -0,0 +1,375 @@
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property
+      - Type
+      - Description
+    * - partitionSpecialCases
+      - ``List<PartitionSpecialCase>`` of `PartitionSpecialCase object`_
+      - A list of partition special cases - endpoints for a partition that do not follow the
+        standard patterns.
+    * - regionSpecialCases
+      - ``List<RegionSpecialCase>`` of `RegionSpecialCase object`_
+      - A list of regional special cases - endpoints for a region that do not follow the
+        standard patterns.
+
+Conflicts with
+    :ref:`aws.endpoints#nonRegionalizedEndpoints-trait`
+
+Most AWS services are regionalized and are strongly encouraged to follow
+the standard endpoint patterns defined above, both for consistency and to
+ensure that endpoints are forwards compatible and that SDK updates are
+not required when the service launches in a new region or partition.
+
+The following example defines a service that uses the standard regional endpoints:
+
+.. code-block:: smithy
+
+    $version: "2"
+
+    namespace smithy.example
+
+    use aws.endpoints#standardRegionalEndpoints
+
+    @standardRegionalEndpoints
+    service MyService {
+        version: "2020-04-02"
+    }
+
+While services are strongly encouraged to follow standard endpoint patterns,
+there are occasional exceptions and special cases.  The following example defines
+a service that use standard regional endpoints, but uses a non-standard pattern for
+FIPS endpoints in US GovCloud:
+
+.. code-block:: smithy
+
+    @standardRegionalEndpoints{
+        partitionSpecialCases: [
+            {
+                partition: "aws-us-gov",
+                endpoint: "myservice.{region}.{dnsSuffix}",
+                fips: true
+            }
+        ]
+    }
+    service MyService {
+        version: "2020-04-02"
+    }
+
+---------------------------
+PartitionSpecialCase object
+---------------------------
+
+A PartitionSpecialCase object contains the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property name
+      - Type
+      - Description
+    * - partition
+      - ``string``
+      - **Required**. The partition to special case (example: "aws").
+    * - endpoint
+      - ``string``
+      - **Required**. The special cased endpoint template.
+    * - dualStack
+      - ``boolean``
+      - When ``true`` the special case will apply to dualstack endpoint variants.
+    * - fips
+      - ``boolean``
+      - When ``true`` the special case will apply to fips endpoint variants.
+
+------------------------
+RegionSpecialCase object
+------------------------
+
+A RegionSpecialCase object contains the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property name
+      - Type
+      - Description
+    * - region
+      - ``string``
+      - **Required**. The region to special case (example: "us-west-2").
```

Comment:
Would it be beneficial to have this as a map key instead of as a member of an object in a list?

Reply:
See comment on PartitionSpecialCase - either way these two should be consistent I think.